### PR TITLE
More caching, now with Invalidation™

### PR DESCRIPTION
**Pull request by andrewconner**
@camhashemi

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala [hunk 1]
```diff
@@ -431,11 +431,8 @@ class TypeaheadCommander @Inject() (
     }
   }
 
-  private def suggestResults(userId: Id[User], limitOpt: Option[Int], dropOpt: Option[Int], requested: Set[TypeaheadRequest]): Seq[TypeaheadSearchResult] = {
-    if (!dropOpt.exists(_ > 0)) { prefetchTypeaheads(userId) } // side effects, preloads result into cache
-
-    val drop = dropOpt.map(Math.min(_, maxSuggestHistory)).getOrElse(0)
-    val limit = limitOpt.map(Math.min(_, maxBatchSize)).getOrElse(10)
+  private def suggestResults(userId: Id[User], limit: Int, drop: Int, requested: Set[TypeaheadRequest]): Seq[TypeaheadSearchResult] = {
+    if (drop == 0) { prefetchTypeaheads(userId) } // side effects, preloads result into cache
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala]: remember that when we provide a `keepId` to `KeepCommander.suggestRecipients`, `drop` always equals `0`

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala]: Ah, good call.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala]: Just saw we're using `prefetchTypeaheadConsolidator`, so we should be fine.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala]: Moved

**Pull request merged**